modules/vpc-sc/main.tf

```
@@ -19,6 +19,35 @@ locals {
     google_access_context_manager_access_policy.default[0].name,
     var.access_policy
   )
+
+  # collect project ids and convert them to numbers
+  _all_project_identifiers = distinct(flatten([
+    for k, v in local.perimeters : [
+      try(v.status.resources, []),
+      try(v.spec.resources, []),
+      [
+        for k, v in local.ingress_policies : [
+          try(v.from.resources),
+          try(v.to.resources)
+        ]
+      ],
+      [
+        for k, v in local.egress_policies : [
+          try(v.from.resources),
+          try(v.to.resources)
+        ]
+      ],
+    ]
+  ]))
+  _project_ids = toset([
+    for x in local._all_project_identifiers :
+    x
+    if can(regex("^projects/[a-z]+", x))
```

```suggestion
    if can(regex("^projects/[a-z]", x))
```
Is enough and more explicit about the intent.